src/providers/postgres/qgspostgresconn.cpp

```diff
@@ -418,11 +424,13 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
   {
     QString errorMsg = PQerrorMessage();
     PQfinish();
+    logWrapper->setError( errorMsg );
     QgsMessageLog::logMessage( tr( "Connection to database failed" ) + '\n' + errorMsg, tr( "PostGIS" ) );
     mRef = 0;
     return;
   }
 
+  logWrapper = nullptr;
```

Comment:
```suggestion
```

this is not required

Reply:
this is necessary for the correct calculation of the execution time...

Reply:
Ok, I didn't notice the rest of the function.